backend/networkTree.js

```diff
@@ -0,0 +1,149 @@
+class Queue {
+
+    constructor() { this.list = []; }
+
+    isEmpty() { return this.list.length === 0; }
+    enqueue(elem) { this.list.push(elem); }
+    dequeue() { if(!this.isEmpty()) { return this.list.shift(); } }
+}
+
+class networkNode {
+    constructor(socketID, maxClients) {
+        this.socketID = socketID;
+        this.maxClients = maxClients;
+    }
+}
+
+class networkTree {
+
+    constructor(socketID, maxClients) {
+        this.node = new networkNode(socketID, maxClients);
+        this.subNodes = [];
```

Comment:
As per the Tree data structure's terminology, maybe children or childNodes would work better here.